examples/self_supervised_learning/losses/_wav2vec2_loss.py

```diff
@@ -0,0 +1,84 @@
+from typing import Optional, Tuple
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+
+def compute_contrastive_loss(
+    x: Tensor,
+    mask_indices: Tensor,
+    targets: Tensor,
+    neg_is_pos: Tensor,
+    reduce: bool,
+    logit_temp: float = 0.1):
+    """
+    Computes the contrastive loss used in Wav2Vec2 loss function.
+
+    Args:
+        x (Tensor): Input embeddings of shape (batch_size, sequence_length, hidden_size)
+        mask_indices (Tensor): Indices to mask negative samples
+        targets (Tensor): Labels indicating positive samples of shape (batch_size, sequence_length, hidden_size).
+        neg_is_pos (Tensor): Boolean tensor indicating whether negative samples should be treated as positives
+        reduce (bool): Whether to reduce the loss to a scalar or not.
+        logit_temp (float, optional): Temperature scaling factor for logits, defaults to 0.1
+
+    Returns:
+        The computed contrastive loss and sample size
+    """
+
+    x = (
+        x[mask_indices]
+        .view(x.size(0), -1, x.size(-1))
+        .unsqueeze(0)
+        .expand(targets.shape)
+    )
+    logits = torch.cosine_similarity(x.float(), targets.float(), dim=-1).float()
+    logits /= logit_temp
+    if neg_is_pos.any():
+        logits[1:][neg_is_pos] = float("-inf")
+    target = logits.new_zeros(logits.size(1) * logits.size(2), dtype=torch.long, device=logits.device)
+    logits = logits.transpose(0, 2)
+    logits = logits.reshape(-1, logits.size(-1))
+    loss = F.cross_entropy(
+        logits,
+        target,
+        reduction="sum" if reduce else "none",
+    )
+    sample_size = target.numel()
+    return loss, sample_size
+
+def wav2vec2_loss(
+    x: Tensor,
+    mask_indices: Tensor,
+    positives: Tensor,
+    negatives: Tensor, 
+    reduce: Optional[bool]=True
```

Comment:
we can simply the `reduce` argument, for example, rename it to `reduction`  with `str` type and only accept `"sum"` and `"none"`.